File: Backend/app/models/health.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey
from app.database.database import Base

class HealthMetrics(Base):
    __tablename__ = "health_metrics"

    id = Column(Integer, primary_key=True, index=True)
    target_type = Column(String)  # "district", "village", "field"
    target_id = Column(String)
    health_score = Column(Integer)
    disease_risk = Column(Integer)
    water_stress = Column(Integer)
    yield_prediction = Column(Float)
    date = Column(String)

class YieldPrediction(Base):
    __tablename__ = "yield_predictions"

    id = Column(Integer, primary_key=True, index=True)
    target_type = Column(String)  # "district", "village", "field"
    target_id = Column(String)
    predicted_yield = Column(Float)
    confidence = Column(Float)
    date = Column(String)

class Alerts(Base):
    __tablename__ = "alerts"

    id = Column(Integer, primary_key=True, index=True)
    severity = Column(String)  # "info", "warn", "alert", "good"
    title = Column(String)
    message = Column(String)
    target_type = Column(String)  # "district", "village", "field"
    target_id = Column(String)
    date = Column(String)

File: Frontend/src/types/Health.ts
export interface HealthSummary {
  rice_bowl_health_index: string;
  average_ndvi: number;
  healthy_area: string;
  disease_risk_area: string;
  expected_yield: string;
}

export interface Alert {
  id: number;
  severity: "info" | "warn" | "alert" | "good";
  title: string;
  message: string;
  target_type: string;
  target_id: string;
  date: string;
}

export interface VillageComparison {
  id: number;
  name: string;
  district: string;
  ndvi: number;
  health: number;
  diseaseRisk: string;
  waterStress: string;
  harvestReady: string;
  yieldPred: string;
}
